### DISCUSSION TITLE
Xdrip in focus after phone unlock

### DISCUSSION THREAD

**Discussion by Schoggiweggli** [NightscoutFoundation/xDrip#3384] [Q&A]
Hey, i have an annoying behavior with Xdrip AFTER an alert goes off. When i unlock the phone Xdrip is automatically displayed and not the homescreen or whatever else. How can this be turned off ? is this the supposed behavior ? Thanks.

I am on version: 3e86af7-2023.11.14 / Android 14

**Comment by Navid200** [comment]
I attempted to test this.  It did not do what you describe.

I set up xDrip as a Nightscouot follower.  Made sure that it was working fine.
Enabled missed reading alert in xDrip.
I then disabled WiFi and data on the phone.
I opened the calculator and locked the phone.

When the missed reading alert triggered, I unlocked the phone.  It came up with the calculator showing, not xDrip.

So, I'm not sure why you experience what you describe.

**Comment by Schoggiweggli** [comment]
I tested what you described with the misssed reading alert and you are right, it doesn't happen then. 
I also use Nightscout follower.

However it happens every time with LOW or HIGH alerts. They seem to behave different ? it's weird.

**Reply by Navid200** [reply]
Thanks for testing that.
I can suspect that is intentional.  Is it a problem?

**Comment by Schoggiweggli** [comment]
It is a mild annoyance. I was confused because years ago when i was using xDrip i didn't recall this behavior (could be wrong). Nor does any other app that i used behave in this way.
It also laggs out the unlock animation quite a bit on my end.

An option to turn this off for High, Low alterts would be nice to have a choice. If that's indeed intended.
I usually dismiss the alert from the lockscreen notification so there's no point having xDrip loaded up after unlocking later.

**Reply by Navid200** [reply]
Let me test.

**Reply by Navid200** [reply]
I can confirm that after a high alert, xDrip was showing after unlocking the phone.
I cannot speak for the developer who did this.
But, I can imagine it is a safety measure.

If I'm sleep and an alert wakes me, I'd like to see what's going on, how fast it is changing, ...
So, after unlocking the phone, I don't want to also have to look for the app.

Is this really such an inconvenience that you are asking for an option?
There are just so many options already in xDrip.

**Reply by Navid200** [reply]
Has this ever resulted in a malfunction?

For example, did you ever have an email compose screen open, locked the phone without sending the message, xDrip alert went off, unlocked the phone, attended to the alert, went back to the email screen and the unsent message was lost?

**Reply by Schoggiweggli** [reply]
Thanks for confirmation.

Not really. It's a mild annoyance, for me personally. I was confused because i never experienced any other app with this behavior (Dexcom, Libre, Guardian, Juggluco, Diabox etc.) so i thought that it is maybe a bug.

I feed xDrtip through Juggluco so i turn the alerts off in xDrip and use the ones on Juggluco for the time as a workaround.
